src/python/foglamp/sending_process.py

```diff
@@ -0,0 +1,813 @@
+_PLUGIN_TYPE = "translator"
+
+_DATA_SOURCE_READINGS = "readings"
+_DATA_SOURCE_STATISTICS = "statistics"
+_DATA_SOURCE_AUDIT = "audit"
+
+# FIXME: set proper values
+_CONFIG_DEFAULT = {
+    "enable": {
+        "description": "A switch that can be used to enable or disable execution of the sending process.",
+        "type": "boolean",
+        "default": "True"
+    },
+    "duration": {
+        "description": "How long the sending process should run before stopping.",
+        "type": "integer",
+        "default": "120"
+    },
+    "source": {
+        "description": "Defines the source of the data to be sent on the stream, "
+                       "this may be one of either readings, statistics or audit.",
+        "type": "string",
+        "default": _DATA_SOURCE_READINGS
+    },
+    "blockSize": {
+        "description": "The size of a block of readings to send in each transmission.",
+        "type": "integer",
+        "default": "5000"
+    },
+    "sleepInterval": {
+        "description": "A period of time, expressed in seconds, "
+                       "to wait between attempts to send readings when there are no readings to be sent.",
+        "type": "integer",
+        "default": "1"
+    },
+    "translator": {
+        "description": "The name of the translator to use to translate the readings "
+                       "into the output format and send them",
+        "type": "string",
+        "default": "omf_translator_new"
+        # FIXME:
+        # "default": "omf_translator"
+    },
+
+}
+_CONFIG_CATEGORY_NAME = 'SEND_PR'
+_CONFIG_CATEGORY_DESCRIPTION = 'Configuration of the Sending Process'
+
+# Configurations retrieved from the Configuration Manager
+_config_from_manager = ""
+
+# Configurations used in the Sending Process
+_config = {
+    'enable': _CONFIG_DEFAULT['enable']['default'],
+    'duration': int(_CONFIG_DEFAULT['duration']['default']),
+    'source': _CONFIG_DEFAULT['source']['default'],
+    'blockSize': int(_CONFIG_DEFAULT['blockSize']['default']),
+    'sleepInterval': int(_CONFIG_DEFAULT['sleepInterval']['default']),
+    'translator': _CONFIG_DEFAULT['translator']['default'],
+}
+
+# Plugin handling - loading an empty plugin
+_module_template = _TRANSLATOR_PATH + "empty_translator"
+_plugin = importlib.import_module(_module_template)
+_plugin_info = {
+    'name': "",
+    'version': "",
+    'type': "",
+    'interface': "",
+    'config': ""
+}
+
+# DB references
+_DB_CONNECTION_STRING = 'postgresql:///foglamp'
+_pg_conn = ()
+_pg_cur = ()
+
+_logger = ""
+_event_loop = ""
+_stream_id = 0
+
+
+class PluginInitialiseFailed(RuntimeError):
+    """ PluginInitializeFailed """
+    pass
+
+
+class UnknownDataSource(RuntimeError):
+    """ the data source could be only one among: readings, statistics or audit """
+    pass
+
+
+class InvalidCommandLineParameters(RuntimeError):
+    """ Invalid command line parameters, the stream id is the only required """
+    pass
+
+
+def _performance_log(func):
```

Comment:
I will check with Mark

Reply:
It would be nice to have a global solution to measuring performance, probably via the continuous integration initiative. Until then let's leave this stuff here so that we have something and review once we have the global method.